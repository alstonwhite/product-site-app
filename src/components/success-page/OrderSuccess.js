import React, { useEffect } from 'react';

import { useDispatch } from "react-redux";
import { clearCartR } from '../../redux/actions';

const OrderSuccess = () => {

    const dispatch = useDispatch();
    useEffect(() => dispatch(clearCartR()), []);
    
    return (
        <div className="success-page" id="success-page">
            <div className="success-page__spacer"></div>
            <h2>Your order has been placed!</h2>
        </div>
    );
};
    
export default OrderSuccess;