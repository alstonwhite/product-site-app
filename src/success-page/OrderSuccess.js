import React, {useState, useEffect} from 'react';

import { useDispatch } from "react-redux";
import { clearCartR } from '../redux/actions';

const OrderSuccess = () => {

    const dispatch = useDispatch();
    useEffect(() => dispatch(clearCartR()), []);
    
    return (
        <div className="success-page" id="success-page">
            <h2>Your order has been placed!</h2>
        </div>
    );
};
    
export default OrderSuccess;