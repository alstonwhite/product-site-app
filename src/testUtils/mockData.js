export const testSections = 
    [
        {
            id: "0",
            slug: "section1",
            sectionTitle: "Section 1",
            active: true
        },
        {
            id: "1",
            slug: "section2",
            sectionTitle: "Section 2",
            active: true
        }
    ]

export const testProducts = 
    [
        {
            id: "0",
            image1: {fields: {file: {url: "1"}}},
            image2: {fields: {file: {url: "2"}}},
            image3: {fields: {file: {url: "3"}}},
            price: 5,
            title: "Product 1",
            description: "Description",
            productSection: {fields: {slug: "section1"}},
            active: true
        },
        {
            id: "1",
            image1: {fields: {file: {url: "1"}}},
            image2: {fields: {file: {url: "2"}}},
            image3: {fields: {file: {url: "3"}}},
            price: 5,
            title: "Product 2",
            description: "Description",
            productSection: {fields: {slug: "section1"}},
            active: true
        },
        {
            id: "2",
            image1: {fields: {file: {url: "1"}}},
            image2: {fields: {file: {url: "2"}}},
            image3: {fields: {file: {url: "3"}}},
            price: 5,
            title: "Product 3",
            description: "Description",
            productSection: {fields: {slug: "section2"}},
            active: true
        }
    ]

export const testProduct = 
    {
        id: 0,
        image1: {fields: {file: {url: "1"}}},
        image2: {fields: {file: {url: "2"}}},
        image3: {fields: {file: {url: "3"}}},
        price: 5,
        title: "Product 1",
        description: "Description",
        productSection: {fields: {slug: "section1"}},
        active: true
    }

export const testCartProduct = 
[{
    id: 0,
    image1: {fields: {file: {url: "1"}}},
    image2: {fields: {file: {url: "2"}}},
    image3: {fields: {file: {url: "3"}}},
    price: 5,
    title: "Product 1",
    description: "Description",
    productSection: {fields: {slug: "section1"}},
    active: true,
    quantity: 1
}]
