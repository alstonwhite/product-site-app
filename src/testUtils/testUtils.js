import React from "react";
import { Router, Route } from "react-router-dom";
import { render } from "@testing-library/react";
import { createMemoryHistory } from "history";

export const renderWithRouter = (ui, 
    {
        path="/",
        route="/",
        history = createMemoryHistory({ initialEntries: [route] }) 
    }={}) => ( 
    render( 
        <Router history={history}> 
            <Route path={path} children={ui} /> 
        </Router> 
    ) 
)