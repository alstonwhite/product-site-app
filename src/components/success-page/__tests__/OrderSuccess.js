import React from 'react'
import { cleanup } from '@testing-library/react'
import OrderSuccess from '../OrderSuccess'
import { renderWithRouter } from '../../../testUtils/testUtils'


afterEach(cleanup)

const mockDispatch = jest.fn();
jest.mock('react-redux', () => ({
    useDispatch: () => mockDispatch
}));

test('calls cart clear on render', () => {
    renderWithRouter(<OrderSuccess/>, {path: "/order-confirm", route: "/order-confirm"})

    expect(mockDispatch).toHaveBeenCalledTimes(1)
    expect(mockDispatch).toHaveBeenCalledWith({
        type: 'CART_CLEAR'
    })
})

test('snapshot', () => {
    const {container} = renderWithRouter(<OrderSuccess/>, {path: "/order-confirm", route: "/order-confirm"})
    expect(container.firstChild).toMatchSnapshot()
  })