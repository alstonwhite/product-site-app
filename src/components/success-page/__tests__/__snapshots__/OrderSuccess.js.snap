// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  class="success-page"
  id="success-page"
>
  <div
    class="success-page__spacer"
  />
  <h2>
    Your order has been placed!
  </h2>
</div>
`;
